### ISSUE TITLE
SequencePAR中的padding策略和论文中描述的不一致

### ISSUE THREAD

**Issue by Bentonmaster** [Event-AHU/OpenPAR#64]
SequencePAR中提供的数据处理方法是先进行resize再进行padding，再对padding后的图像裁剪到224的尺寸，但是根据论文中的描述，使用的方法是先补padding到正方形再进行resize。我应该遵循哪种方法

**Comment by Bentonmaster** [comment]
我在PETA数据集中进行了复现，得到的结果和论文差距较大，我使用了遵循论文描述的padding方法，并使用PromptPAR/dataset/preprocess/peta_pad.py来生成pad.pkl文件（SequencePAR中没有提供），训练参数如下，请问我应该如何做能够复现论文中的结果

> OrderedDict([('dataset', 'PETA'),
             ('batchsize', 32),
             ('epoch', 50),
             ('height', 224),
             ('width', 224),
             ('lr', 1e-06),
             ('weight_decay', 0.0001),
             ('optim', 'AdamW'),
             ('train_split', 'trainval'),
             ('valid_split', 'test'),
             ('gpus', '6'),
             ('redirector', True),
             ('epoch_save_ckpt', 1),
             ('check_point', False),
             ('dir', None),
             ('depth', 6),
             ('beam_size', 1),
             ('pkl_path', None),
             ('datasets_path',
              '/SequencePAR/custom/Pad_datasets/PETAPad/'),
             ('model_path',
              'SequencePAR/custom/model'),
             ('use_class_weight', False)])
  train set: PETA trainval, test set: test
  trainval set: 11400, test set: 7600, attr_num : 35
  max_length 36, vocab_size 38, decoder depth 6
  The optimizer used AdamW

训练结果

> ----------------------2025-10-30_07_51_08 on train set----------------------
 ma: 0.7520, Acc: 0.5061, F1: 0.6658
----------------------2025-10-30_07_59_18 on Evalution set,valid_loss:----------------------
 ma: 0.8350,  pos_recall: 0.7597 , neg_recall: 0.9104 
 Acc: 0.7697, Prec: 0.8314, Rec: 0.8882, F1: 0.8558
------------------------------------------------------------
best ma is 0.8355, in epoch 43 best f1 is 0.8562, in epoch 41

**Comment by NOP1224** [comment]
其次 我们论文中设置的学习率为1e-5

**Comment by Bentonmaster** [comment]
> 代码中的padding是数据增强，而不是论文中提到的Padding操作，论文中Padding操作时时对图片预处理，黑边填充为正方形。所以数据处理与论文描述一致

是的，我的疑问是你们readme中说明可以使用attrdataset.py中的fill code来处理原始数据集，这个部分会造成困扰。我现在是遵循论文中的描述来进行图片的预处理的

**Comment by NOP1224** [comment]
稍等我们会上传PETA的模型，观察你的实验结果确实有异常，训练集我们在50Epoch一般ma都在99+%

**Comment by NOP1224** [comment]
通过网盘分享的文件：PETA_best.pth
链接: https://pan.baidu.com/s/1vQzffLCk-pHOr9gUPm7tBw?pwd=74p7 提取码: 74p7 
--来自百度网盘超级会员v5的分享

**Comment by NOP1224** [comment]
请将 line 57
model.load_state_dict(checkpoint['model_state_dict'],strict=False)  strict设置为True 
似乎是因为我更新了base_block.py导致键名不匹配
